src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java

```diff
@@ -161,14 +147,14 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
 
         // The build should not fail if no merge is expected
         if (intendToMerge && canMerge && (isMergeable == null || !isMergeable)) {
-            logger.println("Pull request cannot be automerged.");
+            listener.getLogger().println("Pull request cannot be automerged.");
             commentOnRequest("Pull request is not mergeable.");
-            listener.finished(Result.FAILURE);
-            return false;
+            listener.error(Result.FAILURE.toString());
```

Comment:
My understanding is `.getLogger()` returns the `PrintStream` object that will receive the output of the build. The `.error()` method returns a `PrintWriter` object that is `A writer to receive details of the error` according to the docs. My guess is Jenkins treats these differently somewhere upstream, but I'm not certain.